libdnf/module/modulemd/ModuleMetadata.cpp

```diff
@@ -193,4 +193,32 @@ void ModuleMetadata::reportFailures(const GPtrArray *failures)
     }
 }
 
+ModulemdObsoletes * ModuleMetadata::getNewestActiveObsolete(ModulePackage *p)
+{
+    ModulemdModule * myModule = modulemd_module_index_get_module(resultingModuleIndex, p->getName().c_str());
+    if (myModule == nullptr) {
+        return nullptr;
+    }
+
+    GError *error = NULL;
+    ModulemdModuleStream * myStream = modulemd_module_get_stream_by_NSVCA(myModule,
+                                                                          p->getStreamCStr(),
+                                                                          p->getVersionNum(),
+                                                                          p->getContextCStr(),
+                                                                          p->getArchCStr(),
+                                                                          &error);
+    if (error) {
+        auto logger(libdnf::Log::getLogger());
```

Comment:
When the error appears? When I ask for data that are not present? Like I have module `nodejs`, stream `12` but no obsolete data available. Or when I ask for data for stream that doesn't exist at all? Like stream `sssss`.

Reply:
When you ask for a stream that doesn't exists. It shouldn't happen.